thirdeye/thirdeye-frontend/app/pods/components/self-serve-alert-yaml-details/component.js

```diff
@@ -38,6 +39,15 @@ export default Component.extend({
     }
   },
 
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    const { alertData: { metric: metrics = [] } = {} } = this;
+
+    this.set('firstMetric', metrics[0]);
+    this.set('remainingMetricCount', metrics.length - 1);
```

Comment:
if the metrics is empty, the count should be 0, but it shows -1.

Reply:
This is remaining metric count, not the actual metric count. We display the first metric and the count of remaining metrics. That is the reason you subtract by 1. If there are no metrics or just 1 metric then the remaining metric count will not be displayed in `hbs` as the `hbs` has the condition to display the count only if the count is greater than 0.

Reply:
Ok, that's great.